### PR TITLE
feat(tournament): game-format label + "automatic + N" rounds (lobby protocol v7)

### PR DESCRIPTION
**Pull request by rykerwilliams**
Adds two organizer-facing options to the tournament create flow, bumping the lobby protocol **6→7**. Both are broker-authoritative (the tournament is shared, broadcast state) and **purely additive**, so `MIN_SUPPORTED_LOBBY_PROTOCOL` stays at 2 and — unlike v6's scoring relaxation — **no capability floor is needed**: a pre-v7 broker ignores the new client fields (a silent capability loss, not a parse error), and its summary omitting `format` is inert against a `JSON.parse` client.

## What it adds

- **Game format** — `CreateTournament.format: Option<GameFormat>`, a display label only (the tournament enforces no deck legality and never touches `GameState`), echoed back resolved on `TournamentSummary.format`. Mirrors the `format: Option<GameFormat>` a `LobbyGame` listing already carries. The create form renders a `<select>` from the shared `FORMAT_REGISTRY` (the same list the game-host picker uses); the detail page shows the resolved label.
- **Automatic + N rounds** — `CreateTournament.plus_rounds: Option<u32>` adds N to the bracket/arity-derived default round count (the "Swiss plus N" community shape). The resolver folds it into the live-default tier, and the round-1 latch captures `base + N`, so the "+N" survives a mid-event drop exactly as the base does. Mutually exclusive with the exact `total_rounds` override — rejected at validation.

## Mirror kept in lockstep

The tournament protocol is mirrored across `lobby-broker` and `server-core`, so both `LobbyClientMessage::CreateTournament` and `ClientMessage::CreateTournament` gain the two fields in the **same declaration order**, and `phase-server`'s projection threads them. The version bumped atomically across Rust (`LOBBY_PROTOCOL_VERSION`), TS (`ws-adapter.ts`), and `scripts/check-protocol-version.mjs`; the version-history doc entry and the `spans_lobby_versions_four_through_seven` test grew a step per the file's stated convention.

## Verification

- **Rust**: `cargo clippy` (lobby-broker + server-core + phase-server) clean; lobby-broker 221 lib/integration tests pass (incl. new resolver/latch, format-projection, and mutual-exclusion tests); phase-server both canonical mirror-roundtrip tests pass (the two enums *and* the view-with-`format` serialize identically across the projection).
- **Frontend**: type-check, eslint, 255 tests (incl. wire-frame, form-submission, and locale-parity — 4 new keys × 7 locales), and `check-protocol-version.mjs` all green.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_011BvCePEHnBgxPwsDPdYupM

## Summary by CodeRabbit

* **New Features**
  * Tournament creation now supports selecting an optional game format.
  * Added an “Automatic + N” option for configuring additional Swiss rounds.
  * Tournament details display the selected game format.
* **Bug Fixes**
  * Prevented conflicting fixed-round and additional-round settings from being submitted together.
* **Documentation**
  * Added localized labels and guidance for the new settings in English, German, Spanish, French, Italian, Polish, and Portuguese.

### CODE REVIEW

**Review by matthewevans — CHANGES_REQUESTED** [review]
**Request changes — one blocking rendering defect with a two-line fix, plus two trivial cleanups worth folding into the same push. Reviewed at head `a97cc94ba8122100749b7066ff8797fad2b478eb`.**

The architecture here is right, and several parts of it are better than they needed to be — details at the end. The blocker is a single guard that tests the wrong absence sentinel.

## 🔴 Blocking

### An empty format pill renders on the default no-format path — `client/src/pages/TournamentPage.tsx:417`

The badge is guarded on `undefined`, and the wire sends `null`:

```tsx
{view.summary.format !== undefined && (
  <span className="rounded-[5px] border border-sky-300/20 bg-sky-500/15 px-1.5 py-0.5 font-semibold text-sky-200">
    {formatMetadata(view.summary.format)?.label ?? view.summary.format}
  </span>
)}
```

With `format: null`, the guard passes, `formatMetadata(null)` returns undefined, and the `??` falls through to `null`. React renders nothing inside a span that still carries a border, padding, and a background fill — an empty sky-coloured pill beside the status and bracket badges.

The wire value is `null`, verified end to end at this head:

- `crates/lobby-broker/src/protocol.rs:789-790` declares `#[serde(default)] pub format: Option<GameFormat>` with **no `skip_serializing_if`**. That attribute appears 20 times elsewhere in the same file, so its absence here reads as deliberate, and `None` therefore serializes as `"format": null` rather than being omitted.
- The PR's own test pins exactly that. `client/src/services/__tests__/tournamentClient.test.ts:397` asserts the outbound frame ends `...,"total_rounds":null,"plus_rounds":null,"format":null}}`, under a comment at `:395` reading "`Option<u32>` with `#[serde(default)]` and no `skip_serializing_if`."

It is the default path rather than an edge case. `CreateTournamentForm.tsx:50` initialises `useState<GameFormat | "">("")`, `:174` renders `create.formatNone` as the first `<option value="">`, and `:107` submits `format: format === "" ? null : format`. An organizer who never opens the format dropdown gets the empty pill.

Two things kept this invisible:

- **The type is inaccurate to the wire.** `client/src/adapter/types.ts:4807` declares `format?: GameFormat` with no `| null`, which is what lets `!== undefined` narrow cleanly and satisfy `tsc`. Its doc comment states "`undefined` when the organizer named none," which is the same assumption in prose.
- **No test exercises the field.** `client/src/pages/__tests__/TournamentPage.test.tsx` contains zero occurrences of `format`; the fixture omits it, producing `undefined` — the one value the guard handles correctly.

The codebase already has the correct shape for the mirrored field. `client/src/components/lobby/GameListItem.tsx:72` consumes `LobbyGame.format` as `game.format ?? "Standard"`, and `LobbyGame.format` at `protocol.rs:628` has the identical `#[serde(default)] Option<GameFormat>` declaration with no `skip_serializing_if`.

**Fix:**

```diff
- // types.ts:4807
- format?: GameFormat;
+ format?: GameFormat | null;

- // TournamentPage.tsx:417
- {view.summary.format !== undefined && (
+ {view.summary.format != null && (
```

The general rule is worth stating, since this seam will recur every time a lobby field is added: a Rust `Option` field **without** `skip_serializing_if` arrives as `null`, not as a missing key. `field?: T` models only the missing-key encoding, so it under-describes that wire; `field?: T | null` models both, and `!= null` is the guard that covers both. `undefined` alone is only correct for a field that is genuinely omitted — which here is just the pre-v7 broker case, not the common one.

## 🟡 Non-blocking

**1. The client-direction round-trip fixture cannot fail — `crates/phase-server/src/main.rs:15004-15005`.** `tournament_client_frames` builds its `CreateTournament` with `plus_rounds: None, format: None`, so `tournament_variants_survive_the_canonical_lobby_roundtrip` would pass identically against a projection that hardcoded `format: None` instead of forwarding `format: *format` — both serialize to `"format":null`, and both compile. The PR gets this right in the other direction at `main.rs:15091`, using `format: Some(engine::types::format::GameFormat::Commander)` under a comment that says why: "A concrete label, so the round-trip cannot pass by dropping it." Applying the same treatment to the client-direction fixture is a one-word change.

**2. The version-history comment miscounts and repeats itself — `client/src/adapter/ws-adapter.ts:474`.** It opens "Three fields added to `CreateTournament`" and then names `format`, `plus_rounds`, and `format` again — the third item is the `TournamentSummary` echo, which is a different message. Two fields are added to `CreateTournament`. The Rust twin enumerates the same change correctly.

**3. `plus_rounds` is accepted for `SingleElimination`, which advertises rounds the bracket can never pair.** Calling this out for the record rather than as a request: `total_rounds: Some(99)` with a single-elimination bracket is already accepted on `main` — `create_tournament` rejects only `Some(0)` and leaves the count otherwise unbounded, which reads as a deliberate tournament-policy judgment. This PR inherits that policy rather than introducing it, so it is out of scope here.

## ✅ What is right, specifically

- **The round-count fold is at the correct seam.** `default_total_rounds` has exactly one non-test call site — `tournament.rs:844`, inside `TournamentMeta::total_rounds()` — and `plus_rounds` folds in there rather than beside it. Every other mention in the file is a doc reference or a test.
- **The latch is correct by construction, not by luck.** `tournament.rs:2094` computes `let total_rounds = meta.total_rounds()` and `:2152` latches that same binding into `resolved_total_rounds`. Because both read one local, a base+N count cannot drift if the player count changes mid-event; a read-time-only implementation would have to be written differently to compile. The accompanying test discriminates against that alternative rather than merely covering the happy path.
- **Mutual exclusion is enforced at all three real ingress guards**, not only in a unit test, and the client/server mirrors line up field for field.
- **The backward-compatibility argument holds under inspection.** No `#[serde(deny_unknown_fields)]` attribute exists in either wire crate — the five occurrences of that string in `protocol.rs` are all doc-comment prose reasoning about the concept, including this PR's own version entry at `:447`. A pre-v7 broker ignores the new fields, and a pre-v7 summary omitting `format` is inert against a `JSON.parse` consumer.
- **The version bump is gate-enforced rather than asserted.** `client/package.json:12` defines `protocol:check`, and `:13` runs it inside `type-check`, so the Frontend job at this head is a real check on the constant. Six `EXPECTED_*` constants live in `scripts/check-protocol-version.mjs` — three current-version surfaces (`PROTOCOL_VERSION` 66 full-game, `LOBBY_PROTOCOL_VERSION` 7, `WIRE_PROTOCOL_VERSION` 49 P2P), two frozen capability floors (`..._TOURNAMENT_ACK` 5, `..._DEFAULT_SCORING` 6), and `DIRECTORY_VERSION` 1. The diff to that file is `+1/-1`: `EXPECTED_LOBBY_PROTOCOL_VERSION` 6 → 7, and nothing else moved.
- **Locale parity is exact** — 7 locales × 4 keys, `+4/-0` each.

One qualifier to pass on, inherited rather than introduced by this PR: `GameFormat`'s **manual** `Deserialize` at `crates/engine/src/types/format.rs:182-195` hard-errors on an unknown format string. So while the *field set* is additive-safe, the *value space* is not — a future format name sent to an older peer is a parse error, not an ignored field. `LobbyGame.format` has the same property today, so this changes nothing about the current risk; it is worth knowing before the value set grows.

**Recommendation: request changes.** Fix the 🔴 guard along with 🟡 #1 and #2 in one push — all three are a few characters each. 🟡 #3 is inherited policy and explicitly does not block.

**Comment by matthewevans** [comment]
**Supplement to my review at head `a97cc94ba8122100749b7066ff8797fad2b478eb`** — a maintainer position on the four CodeRabbit inline findings sitting at that same head, so it is clear which are worth your time. **None of the four blocks this PR.** One of them we are overruling outright; please do not implement it.

### B1 — `Number.parseInt("1e2", 10)` returns `1` (`CreateTournamentForm.tsx:97`) — non-blocking; fix both fields or neither

The finding is correct about the behaviour. The scope framing needs one correction, because the tidy version of this ("the PR introduced a parsing weakness") is not what the diff shows: `const parsedRounds = Number.parseInt(roundsInput, 10)` is unmodified context, while `const parsedPlus = Number.parseInt(plusRoundsInput, 10)` is added by this PR. So the new field follows the existing field's pattern consistently. Fixing it properly means fixing both together — a fair small ask, and not a merge condition.

If you do write a test for it, one warning that will save you time: happy-dom normalizes a pasted `1e2` in a number input, so the defect may not reproduce under the frontend test environment even though it is real in a browser.

### B2 — `min-h-[44px]` touch targets (`CreateTournamentForm.tsx:172`) — non-blocking, out of scope here

The convention is real, and worth knowing about for future frontend work:

- `client/src/index.css:251` names "the 44pt touch-target rule" verbatim.
- `client/src/components/board/PermanentCard.tsx:762` and `:1180-1187` reason explicitly about "the 44px touch-target floor," including when it is unreachable and why.
- It is applied at `PreferencesModal.tsx`, `LimitedDeckBuilder.tsx`, `CubeSetupPanel.tsx:194`, and `DraftPodPage.tsx:403`.
- `client/src/components/draft/__tests__/LimitedDeckBuilder.test.tsx:960` pins it — "keeps the 44px coarse-pointer floor on both chip dimensions."

It does not apply to this PR for scope reasons. `CreateTournamentForm.tsx` carries zero `min-h-[44px]` at this head and zero on `origin/main`, so the gap is pre-existing and form-wide. Adding the floor to only the two new controls would leave the form internally inconsistent — the new fields would clear the bar while every control around them still sits under it.

Also note the house idiom is a coarse-pointer floor released on fine pointers, rather than the naked class the bot suggests:

```
min-h-[44px] rounded-[14px] py-2 text-base sm:min-h-0 sm:text-sm            // PreferencesModal.tsx
min-h-[44px] min-w-[44px] … pointer-fine:min-h-0 pointer-fine:min-w-0       // LimitedDeckBuilder.tsx
```

A separate PR applying that idiom across the whole form would be welcome.

### B3 — reject `plus_rounds` for `SingleElimination` (`tournament.rs:845`) — non-blocking; welcome if you are already in the file

The mechanism it names is real. What keeps it off the blocking list is the parity control: `total_rounds: Some(99)` with a single-elimination bracket is **already** accepted on `main` today. The only bound is `tournament.rs:1878` — `if req.total_rounds == Some(0)` — and `validate_create_tournament_fields` says in its own comment that it leaves the count otherwise unbounded on purpose, because "a ceiling would be a tournament-policy judgment, which is `crate::tournament`'s to make." So this PR inherits an existing policy decision rather than introducing a defect.

Since you are already editing `validate_create_tournament_fields`, a `bracket == Swiss` guard would be a few lines in exactly the right place. Offered as a welcome addition, explicitly not as a condition of merge.

### B4 — add an exclusivity check inside `create_tournament` (`tournament.rs:1896`) — overruled, please do not implement

This one we are declining, and it is worth saying why so it does not sit there looking like unaddressed feedback.

Its reachability premise does not hold. Every deserialized frame is validated at the socket boundary before it reaches the manager:

- `crates/lobby-broker/src/protocol.rs:1397-1401` runs `crate::validation::validate_lobby_message(&msg)` on every successfully deserialized `LobbyClientMessage`, mapping a failure to `ParsedFrame::Malformed(reason)`.
- `crates/lobby-broker/src/inbound_guard.rs:201` carries a `_ => validate_lobby_message(msg)?` catch-all, which covers `CreateTournament`.

`validate_create_tournament_fields` already rejects both round fields being set, so a frame carrying both never reaches `TournamentManager::create_tournament`. The only caller that could set both is an in-crate Rust caller hand-building the struct, which is not a system boundary.

Adding the check would run against CLAUDE.md — "Don't add error handling, fallbacks, or validation for scenarios that can't happen… Only validate at system boundaries" — and would create a second authority for a rule `validate_lobby_message` already owns, which is the failure mode that guidance exists to prevent.

---

The blocker and the two nits in my review remain the only merge conditions.

**Comment by matthewevans** [comment]
**Supplements review #5128689500 at head `a97cc94ba8122100749b7066ff8797fad2b478eb`.** Two corrections to that review, and a consolidated ruling on the CodeRabbit findings. **This supersedes my earlier comment on B3** — that one called it non-blocking, and on closer reading it should be a requested change. Reasoning below; the change of position is mine, not something you did.

## Correction 1 — 🟡 #3 is upgraded to a requested change

My review said the SingleElimination + `plus_rounds` combination "inherits that policy rather than introducing it." The first half is right and the second half is wrong, so I am retracting it. The *state* was already reachable through `total_rounds_override`, which is why the parity argument looked sufficient. But `plus_rounds` is new surface in this PR, so this adds a second door into that state. A PR that opens a new route to a bad configuration owns that route, whatever else already led there.

**The codebase also names where the fix belongs, and it is not where CodeRabbit pointed.** The comment above the exclusivity guard in `crates/lobby-broker/src/validation.rs` says of a round ceiling:

> A ceiling would be a tournament-policy judgment, which is `crate::tournament`'s to make.

So this is `crate::tournament`'s call, not the resolver's at `tournament.rs:845`. And the right home already has a sibling doing exactly this job: `create_tournament` at `tournament.rs:1872` rejects `bracket == SingleElimination && arity != HEAD_TO_HEAD`, under a comment reasoning that rejecting explicitly beats silently dropping the request's intent. That is the same shape of decision — a bracket that cannot honour what was asked for — and the same argument applies.

**Requested:** reject `plus_rounds.is_some()` on a `SingleElimination` bracket in `create_tournament`, beside the guard at `:1872`, with a test covering it. Credit to CodeRabbit for identifying the mechanism; the location differs for the reason quoted above.

## Correction 2 — precision on the mutual-exclusion credit

My review credited "mutual exclusion is enforced at all three real ingress guards." That is true as written, and it invites a stronger reading than the evidence supports, so here is the precise picture:

The guard lives at `validation.rs:281` and is enforced on every wire path — `protocol.rs:1397-1401` runs `validate_lobby_message` on every successfully deserialized frame at the socket boundary, mapping failures to `ParsedFrame::Malformed`, and `inbound_guard.rs:201`'s `_ =>` catch-all covers `CreateTournament`. `TournamentManager::create_tournament` itself does **not** re-check: it copies `total_rounds_override` and `plus_rounds` straight through, and if both were somehow set, `total_rounds()` would silently prefer the override.

**This is correct as designed and needs no change.** The only caller that could set both is an in-crate Rust caller hand-building the struct, which is not a system boundary. See B4 below. The correction is about being accurate on what was measured.

## CodeRabbit findings at this head

Four inline findings are open at `a97cc94ba8`. Worth noting for the record: none of them is on `TournamentPage.tsx`, so none found the blocking defect in my review.

**B3 — SE + `plus_rounds` (`tournament.rs:845`) — accepted.** Now folded into the requested changes, per Correction 1, at `create_tournament` rather than the resolver.

**B4 — duplicate exclusivity check in `create_tournament` (`tournament.rs:1896`) — overruled; please do not implement.** The finding is factually right that `create_tournament` has no such check, and wrong that this is a defect. `validation.rs`'s own comment states the design: the contradiction is rejected at the boundary. A second check would create a second authority for a rule `validate_lobby_message` already owns, against CLAUDE.md's "only validate at system boundaries" rule.

The contrast with B3 is the useful part, and it comes from one comment: `validation.rs` assigns *policy* judgments to `crate::tournament` while keeping *contradiction rejection* at the boundary. B3 is a policy judgment, so it moves into `crate::tournament`. B4 is boundary validation that already exists, so it stays put and is not duplicated.

**B1 — `Number.parseInt("1e2", 10)` returns `1` (`CreateTournamentForm.tsx:97`) — non-blocking; fix both fields or neither.** As noted previously: `const parsedRounds = …` is unmodified context while `const parsedPlus = Number.parseInt(plusRoundsInput, 10)` is added by this PR, so the new field follows the existing field's pattern. Fixing it properly means fixing both. If you write a test, happy-dom normalizes a pasted `1e2` in a number input, so it may not reproduce there.

**B2 — `min-h-[44px]` touch targets (`CreateTournamentForm.tsx:172`) — non-blocking, out of scope.** The convention is genuine, and the clearest citation is the sibling file in this same directory: `client/src/components/tournament/PairingsList.tsx:86` calls `min-h-[44px]` "this repo's established spelling" of the ">= 44pt touch-target rule" and applies it at `:98`. `client/src/index.css:251` names the rule, and `LimitedDeckBuilder.test.tsx:960` pins it with an assertion.

It stays out of scope here because the gap is form-wide rather than specific to the new fields. `CreateTournamentForm.tsx` has zero occurrences at this head and zero on `origin/main`, and all nine of its inputs and selects carry a byte-identical `className="rounded-[6px] border border-white/10 bg-black/30 px-3 py-2 text-sm text-gray-100"`. Patching only the two new controls would leave 2 of 9 taller than the rest.

One implementation note for whoever picks this up: the house idiom releases the floor on fine pointers (`sm:min-h-0`, or `pointer-fine:min-h-0`), so the bare class as proposed would also enlarge desktop controls. A follow-up covering all nine in that idiom would be welcome.

---

## Merge conditions, in one place

1. 🔴 `TournamentPage.tsx:417` — guard on `!= null`, and widen `types.ts:4807` to `GameFormat | null`.
2. 🟡 `main.rs:15004-15005` — give the client-direction fixture a concrete `format`, matching `:15091`.
3. 🟡 `ws-adapter.ts:474` — the comment says three fields where two are added, and names `format` twice.
4. 🟡 **new** — reject `plus_rounds` on a `SingleElimination` bracket in `create_tournament`, beside `tournament.rs:1872`, with a test.

B1 and B2 are welcome but not conditions. B4 is declined.

File: client/src/components/tournament/CreateTournamentForm.tsx [hunk 1]
```diff
@@ -132,6 +157,29 @@ export function CreateTournamentForm({
...
+        </label>
+        {/* A display label only — the tournament enforces no deck legality.
+            Options come from the shared FORMAT_REGISTRY (the same list the game
+            host format picker renders), so a new engine format appears here
+            with no change to this component. */}
+        <select
+          id={formatId}
+          value={format}
+          onChange={(event) => setFormat(event.target.value as GameFormat | "")}
+          className="rounded-[6px] border border-white/10 bg-black/30 px-3 py-2 text-sm text-gray-100"
```
[the remark this answers is not in the document]
Comment (matthewevans) [reply client/src/components/tournament/CreateTournamentForm.tsx:172]: Maintainer ruling: non-blocking and out of scope for this PR — full reasoning in https://github.com/phase-rs/phase/pull/8691#issuecomment-5566096983.

The convention is genuine (`PairingsList.tsx:86`, the sibling file in this directory, calls `min-h-[44px]` "this repo's established spelling" of the rule). It does not apply here because the gap is form-wide rather than specific to the new fields: `CreateTournamentForm.tsx` has zero occurrences at this head and zero on `origin/main`, and all nine of its inputs and selects carry a byte-identical className, so patching only the two new controls would leave 2 of 9 taller than the rest. A separate PR covering all nine is welcome — in the house idiom, which releases the floor on fine pointers (`sm:min-h-0` / `pointer-fine:min-h-0`) rather than applying the bare class. Resolving; nothing is being asked of the author here.

File: crates/lobby-broker/src/tournament.rs [hunk 2]
```diff
@@ -1860,6 +1893,8 @@ impl TournamentManager {
                 // Nothing is latched until round 1 is paired: no round has
                 // been scheduled yet, and the field is still empty.
                 resolved_total_rounds: None,
+                plus_rounds: req.plus_rounds,
```
[the remark this answers is not in the document]
Comment (matthewevans) [reply crates/lobby-broker/src/tournament.rs:1896]: Maintainer ruling: overruled — please do not implement this one. Full reasoning in https://github.com/phase-rs/phase/pull/8691#issuecomment-5566096983.

The observation is factually right that `TournamentManager::create_tournament` has no exclusivity check, and the conclusion that this is a defect does not follow. The contradiction is rejected at the boundary: `validation.rs:281` holds the guard, `protocol.rs:1397-1401` runs `validate_lobby_message` on every deserialized frame at the socket boundary, and `inbound_guard.rs:201`'s `_ =>` catch-all covers `CreateTournament`. The only caller that could set both fields is an in-crate Rust caller hand-building the struct, which is not a system boundary. Adding a second check would create a second authority for a rule `validate_lobby_message` already owns, against CLAUDE.md's "only validate at system boundaries" rule. Resolving so this does not read as an outstanding request.

**Comment by rykerwilliams** [comment]
Thanks for the thorough review — pushed `ec3f303ee` addressing all three:

- 🔴 **Empty format pill**: exactly right about the `null`-vs-`undefined` sentinel. Retyped `TournamentSummary.format` as `GameFormat | null` (the accurate wire shape, since the Rust field has no `skip_serializing_if`) and switched the badge guard to `!= null`, which covers both the `null` common path and the pre-v7 `undefined` path. Added the missing `TournamentPage` tests: label renders when present, no pill on `null`, no pill when omitted.
- 🟡 #1: gave the client-direction roundtrip fixture a concrete `format: Some(Commander)` so the projection can't pass by dropping it — matching the server-direction fixture. Both `tournament_variants_survive_the_canonical_lobby_roundtrip` and `..._accepted_in_both_server_modes` pass locally.
- 🟡 #2: fixed the version-history miscount — two fields added to `CreateTournament` (`format`, `plus_rounds`), and the third `format` is the separate `TournamentSummary` echo.

Intentionally left the two you overruled: the 44pt touch-target (form-wide, out of scope) and a create-time exclusivity check (already enforced at `validation.rs:281`). And noted the inherited SingleElimination `plus_rounds` policy + the `GameFormat` manual-`Deserialize` value-space caveat for the record — agreed both are pre-existing and out of scope here.

**Review by matthewevans — APPROVED** [review]
Reviewed at head ec3f303ee9b0297d5d10f6720245189e155012a5

Approving. `ec3f303e` resolves all three findings from my earlier review (which was filed against `a97cc94b`); each is re-verified against the current head, not the review timestamp.

**1. Blocker — empty format pill on the no-format path: resolved.**
`TournamentPage.tsx:417` now guards `{view.summary.format != null && (`, and `adapter/types.ts:4810` is typed `format?: GameFormat | null` — the accurate wire shape, since the Rust field is `#[serde(default)] Option<GameFormat>` with no `skip_serializing_if`, so `None` arrives as an explicit `"format": null` rather than a missing key. `!= null` covers both that common path and the `undefined` a pre-v7 broker sends. The fix is complete, not partial: `summary.format` has exactly one consumer, and no site still tests `undefined`. The doc comment you added on the field carries the reasoning, which is the right place for it.

The three new cases at `TournamentPage.test.tsx:633-667` discriminate. Reverting the guard to `!== undefined` lets `null` through, renders the pill, and reds the `toBeNull()` assertion — and the negative assertions are not vacuous, because the first case proves the `.bg-sky-500/15` selector actually matches when a format is present. That positive reach-guard is what makes the two absence assertions mean something.

**2. Non-blocking — non-discriminating client-direction fixture: resolved.**
`phase-server/src/main.rs:15008` now sets `format: Some(GameFormat::Commander)`, so `tournament_variants_survive_the_canonical_lobby_roundtrip` can no longer pass against a projection that hardcoded `None`. Matches the server-direction fixture's existing treatment.

**3. Non-blocking — miscounted version-history comment: resolved.**
`ws-adapter.ts:474-480` now reads correctly, and states plainly why `MIN_SUPPORTED_SERVER_LOBBY_PROTOCOL` stays at 2 (purely additive in both directions, unlike 6's scoring relaxation).

**Protocol discipline is clean.** All three copies of the lobby constant agree at 7 — `lobby-broker/src/protocol.rs:571`, `ws-adapter.ts:539`, `scripts/check-protocol-version.mjs:10` — against a `main` at 6, and the other three protocol constants are untouched. No competing bump exists in any other open PR. Getting a four-constant cross-language bump right on the first pass is the part of this that most often goes wrong; it didn't here.

---

Two follow-ups, both explicitly **non-blocking** — neither gates this merge.

**A. `plus_rounds: None` in the client-direction fixture (`main.rs:15004`).** This repeats, for `plus_rounds`, the exact test-strength gap you just closed for `format`. It is *not* a live defect: the projection forwards it correctly at `main.rs:4606` (`plus_rounds: *plus_rounds`). It only means the round-trip test would not catch a regression there.

**Do not apply CodeRabbit's suggested remedy for this one** (review 5129420874). It proposes setting `plus_rounds: Some(1)` in place, but that same fixture carries `total_rounds: Some(4)` at `:15003`, and this PR's own validation rejects that pair at `lobby-broker/src/validation.rs:277-283` ("mutually exclusive"). Applying it verbatim would encode a wire-invalid combination into a fixture named *canonical* — which is worse than the gap it closes. The correct shape is a **ninth** frame carrying `total_rounds: None, plus_rounds: Some(2)`, with the `assert_eq!(frames.len(), 8, …)` at `:15156` bumped to 9. Your own comment above that assertion explains why the count is asserted at all, so the bump is part of the remedy, not an afterthought.

**B. `TournamentManager::create_tournament` does not itself reject the both-set contradiction** (`lobby-broker/src/tournament.rs:1859-1880`). Unreachable over the wire — the only non-test caller is `broker.rs:1277`, and validation runs ahead of it at `protocol.rs:1398` and `inbound_guard.rs:201`. CLAUDE.md's "validate at system boundaries, trust internal code" makes the wire boundary the right and sufficient place for this, so this is a note for a future reader rather than a defect. Worth a comment at the manager if it ever grows a second caller.

Required checks are green at this head (`Rust (fmt, clippy, test, coverage-gate)` and `Frontend (lint, type-check, test)`, all four Rust shards). Enqueuing.
